src/components/Counter.js

```diff
@@ -1,59 +1,58 @@
+  };
 
-    incrementAsync = () => {
-        // Stretch Problem: Implement an increment function that
-        // increments after waiting for one second
-    };
+  incrementAsync = () => {
+    // Stretch Problem: Implement an increment function that
+    // increments after waiting for one second
+    setTimeout(() => {
+      this.props.increment();
+    }, 1000);
+  };
 
-    render() {
-        // Fill in the two button onClick methods
-        // Upon clicking these buttons, the count
-        // should decrement or increment accordingly
-        return (
-            <p>
-                Clicked: {this.props.count} times
-                <button onClick={() => /* Fill me in */ }>
-                    +
-                </button>
-                <button onClick={() => /* Fill me in */ }>
-                    -
-                </button>
-                 {/* Uncomment these button tags if you got
+  render() {
+    // Fill in the two button onClick methods
+    // Upon clicking these buttons, the count
+    // should decrement or increment accordingly
+    return (
+      <p>
+        Clicked: {this.props.count} times
+        <button onClick={() => this.props.increment()}>+</button>
+        <button onClick={() => this.props.decrement()}>-</button>
+        {/* Uncomment these button tags if you got
                 around to implementing the extra credit functions */}
-                {/* <button onClick={this.incrementIfOdd}>
-                    Increment if odd
-                </button>
-                <button onClick={this.incrementAsync}>
-                    Increment async
-                </button>  */}
-            </p>
-        );
-    }
+        <button onClick={this.incrementIfOdd}>Increment if odd</button>
+        <button onClick={this.incrementAsync}>Increment async</button>
+      </p>
+    );
+  }
 }
 
-// The mapStateToProps function specifies which portion of the 
-// state tree this component needs to receive. In this case, 
+// The mapStateToProps function specifies which portion of the
+// state tree this component needs to receive. In this case,
 // since our redux store is only storing the value of the count,
 // this component receives the whole state. In a more complex
 // redux application, though, it would receive only the relevant
 // parts it needs from the state object.
-const mapStateToProps = (state) => {
-    return {
-        count: state.count
-    };
+const mapStateToProps = state => {
+  return {
+    count: state.count
+  };
 };
 
 // The connect function is called in order to make this component aware
 // of the rest of the redux architecture. Without this, this component
 // is only a dumb React component. We pass in all of the functions that
 // are reliant on Redux, along with the component itself, so that Redux
 // makes itself known to this component.
-export default connect(mapStateToProps, { increment, decrement })(Counter);
+export default connect(
+  mapStateToProps,
```

Comment:
Do you understand `connect()`? If you have some doubts, I recommend diagramming it out. Try to figure out where everything comes from, as it's one of the most difficult parts of redux to understand.